src/utils/downloadZipFile.ts

```diff
@@ -0,0 +1,83 @@
+import JSZip from 'jszip';
+import sanitize from 'sanitize-filename';
+
+interface ZipFileContent {
+  /**
+   * Name of the file to create inside the zipfile. If there is a suffix specified in the top-level
+   * parameters, it will be appended to this value.
+   */
+  fileName: string;
+
+  /**
+   * Source of the file's content. This can either be the data or a promise that resolves to the
+   * data. If the zipfile contains multiple files whose content comes from promises, they will be
+   * resolved concurrently.
+   */
+  content: string | Blob | (() => Promise<string | Blob | null>);
+}
+
+interface DownloadZipFileParams {
+  /**
+   * Name of the directory inside the zipfile where the files will be placed. Also used as the base
+   * filename of the generated zipfile. For example, if this is `foo`, the generated zipfile will be
+   * called `foo.zip` and a file `xyz.csv` in the zip archive will be called `foo/xyz.csv`.
+   */
+  dirName: string;
+
+  files: ZipFileContent[];
+
+  /** Suffix to append to each file's name. For example, `.csv`. */
+  suffix?: string;
+}
+
+/** Downloads a binary file to the user's system. */
+function downloadBlob(fileName: string, content: Blob) {
+  const url = URL.createObjectURL(content);
```

Comment:
We use `data:` URLs with base64-encoded file contents for file downloads elsewhere in the code. This seemed more robust to me (no concerns about URL length limits, etc.) but happy to change it if this isn't the right approach.